### ISSUE TITLE
ty: assert membership doesn't narrow `str` to `Literal` for `TypedDict` key (minimal repro)

### ISSUE THREAD

**Issue by dsfaccini** [astral-sh/ty#2178]
### Summary
Minimal repro of `assert` not narrowing a `str` to a `Literal` before populating a `TypedDict`.

Playground link: https://play.ty.dev/28bae877-4f82-452c-95c9-5cf5ba8cd0e2

### Repro (no external deps)
`ty_repro_assert_literal.py`:
```python
from typing import Literal, TypedDict
from dataclasses import dataclass

class InputAudio(TypedDict):
    data: str
    format: Literal['wav', 'mp3']

@dataclass
class Item:
    data: str
    format: str

def build_audio(item: Item) -> InputAudio:
    assert item.format in ('wav', 'mp3')
    return InputAudio(data=item.data, format=item.format)

if __name__ == '__main__':
    print(build_audio(Item('x', 'wav')))
```

Run:
```bash
uvx ty check ty_repro_assert_literal.py
```

### Actual result
```
error[invalid-argument-type]: Invalid argument to key "format" with declared type `Literal["wav", "mp3"]` on TypedDict `InputAudio`
  --> ty_repro_assert_literal.py:15:12
   |
13 | def build_audio(item: Item) -> InputAudio:
14 |     assert item.format in ('wav', 'mp3')
15 |     return InputAudio(data=item.data, format=item.format)
   |            ----------                 -------^^^^^^^^^^^
   |            |                          |      |
   |            |                          |      value of type `str`
   |            |                          key has declared type `Literal["wav", "mp3"]`
   |            TypedDict `InputAudio`
```

### Expected result
The membership assert should narrow `item.format` to `Literal['wav', 'mp3']`, allowing assignment into the `TypedDict` without diagnostics.

### Notes
Ty 0.0.4 via `uvx ty check`.

**Comment by MeGaGiGaGon** [comment]
This is the same as #1566

**Comment by dsfaccini** [comment]
aahh sorry for that, I wasn't too diligent in my search, I'll close this one too then.
